lib/modules/manager/gomod/artifacts.ts

```diff
@@ -271,7 +272,9 @@ export async function updateArtifacts({
 
     const execCommands: string[] = [];
 
-    let args = 'get -d -t ./...';
+    const goGetDirs = config.goGetDirs?.map(quote).join(' ') ?? './...';
```

Comment:
Did you find that each path needs to be quoted individually instead of collectively?

We need to check whether it's possible to bypass this quoting by configuring like `["./...", "&&", "cat", "/etc/passwd"]`

Reply:
```
goGetDirs: ['.', 'foo', '.bar/...', '&&', 'cat', '/etc/passwd']
```
will result in 
```
go get -d -t . foo .bar/... '&&' cat /etc/passwd
```
which will crash and not allow the command chaining.

this is what is expected, right?

Reply:
Paging @secustor and @JamieMagee for additional opinions

Reply:
Not sure if this can be defeated by using a combination of single and double quotes as well backslashes. :/

We could check the strings with [`path.parse(path)`](https://nodejs.org/docs/latest-v18.x/api/path.html#pathparsepath) and add one exception for `./...`

Reply:
This user wants it configurable though

Reply:
this allows to pass arbitrary ARGS to go get. so we somehow need to defend it

Reply:
>Not sure if this can be defeated by using a combination of single and double quotes as well backslashes. :/

I mean, it would be easy to remove quotes and backslashes from the input strings, those are not valid filepaths anyways :shrug:

Reply:
Yes, that should be good enough for now. 

Create a utility function for this in https://github.com/renovatebot/renovate/blob/main/lib/util/fs/index.ts and name it `isValidPath`.

Reply:
well, I tried adding such functionality, but that module is mocked in the tests:
```
jest.mock('../../../util/fs',);
```

I tried using `jest.requireActual` to restore functionality but to no avail.
What would be the proper way to "unmock" that function? 

Give me a discount as I have not touched JS in like 2 years :smile:

Reply:
This blog describes how you can setup a mock like you are trying to do: https://www.benmvp.com/blog/mock-all-functions-module-except-one-jest/

But I'm fine with mocking the result of `isValidPath` for gomod, if the examples from your tests are included in the ones for the utility function.